pkg/agent/route/route_linux.go

```diff
@@ -1136,48 +1136,68 @@ func (c *Client) AddClusterIPRoute(svcIP net.IP) error {
 		gw = config.VirtualServiceIPv6
 	}
 
+	// If the route exists and contains the ClusterIP, it's no need to update the route.
+	if curClusterIPCIDR != nil && curClusterIPCIDR.Contains(svcIP) {
+		klog.V(4).InfoS("Route with current ClusterIP CIDR can route the ClusterIP to Antrea gateway", "ClusterIP CIDR", curClusterIPCIDR, "ClusterIP", svcIP)
+		return nil
+	}
+
+	var newClusterIPCIDR *net.IPNet
+	var err error
 	if curClusterIPCIDR != nil {
-		// If the route exists, check that whether the route can cover the ClusterIP.
-		if !curClusterIPCIDR.Contains(svcIP) {
-			// If not, generate a new destination ipNet.
-			newClusterIPCIDR, err := util.ExtendCIDRWithIP(curClusterIPCIDR, svcIP)
-			if err != nil {
-				return fmt.Errorf("extend route CIDR with error: %v", err)
-			}
-			// Generate a new route with new ClusterIP CIDR and install the route.
-			mask, _ = newClusterIPCIDR.Mask.Size()
-			route := generateRoute(newClusterIPCIDR.IP, mask, gw, linkIndex, scope)
-			if err = netlink.RouteReplace(route); err != nil {
-				return fmt.Errorf("failed to install new route: %w", err)
-			}
+		// If the route exists and doesn't contain the ClusterIP, generate a new destination ipNet.
+		newClusterIPCIDR, err = util.ExtendCIDRWithIP(curClusterIPCIDR, svcIP)
+		if err != nil {
+			return fmt.Errorf("extend route CIDR with error: %w", err)
+		}
```

Comment:
nit: could use the more compact syntax `if newClusterIPCIDR, err = util.ExtendCIDRWithIP(curClusterIPCIDR, svcIP); err != nil `